### ISSUE TITLE
Some questions about training set and validation set

### ISSUE THREAD

**Issue by clancylea** [roberthangu/snn_object_recognition#2]
Excuse me，although I see your paper，but i have some question about it.

First,how many images for training set and validation set?

Second,rescaled to be 300 pixels in height (preserving the aspect ratio) ?or others?
For the same size of motorbikes and  faces ,do i need to fill_image?  

Thanks，Forgive me for so many problems. Haha

**Comment by roberthangu** [comment]
Hi @clancylea,
* I used 350 images for training and 50 for validation.
* The images are rescaled such that the **longer edge** is 300 px wide, preserving the aspect ratio. This might be the vertical or horizontal dimension, depending on the image.
* Could you please be more specific about the question regarding fill_image?

**Comment by clancylea** [comment]
> Hi @clancylea,
> 
> * I used 350 images for training and 50 for validation.
> * The images are rescaled such that the **longer edge** is 300 px wide, preserving the aspect ratio. This might be the vertical or horizontal dimension, depending on the image.
> * Could you please be more specific about the question regarding fill_image?

*350 images for training and 50 for validation include motorbikes ,faces and background ?  or motorbike/no motorbike.
*About the question regarding fill_image,if i use motorbikes and faces and background  in training set , because three category of images have different  size. So, if rescaled the longer edge is 300 px wide and preserving the aspect ratio, three category of images may have different size in training set.
*I want to run the svm and one-shot ,  eventually, I can achieve the accuracy of your paper. so i ask these questions.

Thank you very much!

**Comment by roberthangu** [comment]
Hi @clancylea,

* The 350/50 images ratio for training/validation respectively is used for any training you may want to do. It's a general ratio in the experiments. E.g. if you want to train motorbikes, you use only motorbikes and keep the 350/50 ratio. If you want the mixed training scenario, you use mixed images and also keep to the 350/50 ratio. Of course, if you want to train only a certain class, you will use only images of that class. The background training set was used in the mixed scenario to show that the network doesn't get disturbed by the background images.
* Yes, it's correct that the images will have different sizes. They were scaled like this to fit into a square of 300x300 px onto which the network is applied to. Thus the remaining "free space" around the images in that square is black and sends no input spikes. Because the network is by construction scale-invariant (it uses images at different scales) it will learn the features in any case.

I hope my comments help you.

**Comment by ENCoreDong** [comment]
Hi @roberthangu 

Excuse me,i am a student from china ,i have read your paper and i have a question about it.

1. As your said,in the end of learning feature,we get a list containing for each prototype the weights of the connections of the neuron which fired first in the respective prototype.Towards it ,i want to know how can i judge which neuron is fired first .As i can see,in the code ,you choose the weights of the connections of the first neuron of each prototype.

thank you！
![get_cur_weight](https://user-images.githubusercontent.com/43899501/59195432-8bfa6200-8bbe-11e9-8d06-765e066ff74f.png)

**Comment by roberthangu** [comment]
Hi @ENCoreDong,

The weights of the neuron which fires first are copied to all other neurons in that prototype. After the training they all have the same weights, so it doesn't matter which neuron you copy the weights from. That's why I take the first one, because it's arbitrary.

**Comment by ENCoreDong** [comment]
Hi @roberthangu
Excuse me,i am glad to hear your answer,and i also have two questions about  prototype cell in S2.
1.I want to know why we need to create three kinds of  prototype cell with different random i_offsets for stdp in S2  .
 2. why we need to set the different random weights in the synapses between C1 and S2?
As i know,if a week cell in S2 gets a strong  i_offset value,it may become the winner in the competition.
if some strong cells in C2 get weak weights with S2,they may become losers in the competition of stdp.

**Comment by roberthangu** [comment]
Hi @ENCoreDong,

This project is no longer maintained. Please find answers to the questions in my paper, the paper of Masquelier et al. which is referenced in the README on the front page, and in the source code.

Best regards,
Robert

**Comment by roberthangu** [comment]
Closing issue because the project is no longer maintained.
